README.md

```diff
@@ -899,6 +900,66 @@ and need to be removed by hand.
 For long term use, a dedicated project in the Google Cloud Platform is
 recommended to prevent automated manipulation of important machines.
 
+<a name="openstack"/>
+
+## Openstack backend
+
+The Openstack backend is easy to setup and use, and allows distributing
+your tasks to an openstack environment.
+
+_$PROJECT/spread.yaml_
+```
+(...)
+
+backends:
+    openstack:
+        plan: cpu2-ram4-disk10
+        halt-timeout: 2h
+        systems:
+            - ubuntu-20.04:
+                  image: ubuntu-focal-daily-amd64
+                  workers: 5
+
+            # Extended syntax:
+            - another-system:
+                image: some-other-image
+                networks:
+                    - network_external
+                    - network_pvn
+                groups:
+                    - group_external
+
+```
+
+The Openstack backend gets all the information to authenticate from the
+environment variables. The following variables have to be set:
```

Comment:
The most common mechanism  to connect a client through the Openstack API is by sourcing a file with teh environment variables (like the one we get to use openstack client in canonistack). For example in PS5, in the environment I also see the openstack env vars in my environment (they are managed by vault tool), so I presume juju is using those to connect to openstack.

I agree with you this is not a good practice because this means we need to have an env var with the user and password. I'll research which workaround we could use.

Reply:
I updated the documentation to explain better which env vars need to be defined. It is also supported to authenticate by using a key (similar to what we have in google). The key has to be stored in an env var to be loaded.

Reply:
An improvement could be to load the vars from a file (as we hav in google) instead of the env. @niemeyer What do you think?

Reply:
Finally I added a similar approach that the used in google.